aws/resource_aws_vpn_connection.go

```diff
@@ -586,14 +587,20 @@ func waitForEc2VpnConnectionDeletion(conn *ec2.EC2, id string) error {
 	return err
 }
 
-func xmlConfigToTunnelInfo(xmlConfig string) (*TunnelInfo, error) {
+func xmlConfigToTunnelInfo(xmlConfig string, tunnel1_inside_cidr string) (*TunnelInfo, error) {
 	var vpnConfig XmlVpnConnectionConfig
 	if err := xml.Unmarshal([]byte(xmlConfig), &vpnConfig); err != nil {
 		return nil, fmt.Errorf("Error Unmarshalling XML: %s", err)
 	}
 
-	// don't expect consistent ordering from the XML
-	sort.Sort(vpnConfig)
+	// don't expect consistent ordering
+	_, ipnet1, _ := net.ParseCIDR(tunnel1_inside_cidr)
+	if !(ipnet1.Contains(net.ParseIP(vpnConfig.Tunnels[0].VgwInsideAddress))) {
```

Comment:
Please note that this line currently can panic:

```
=== CONT  TestAccAWSVpnConnection_basic

------- Stderr: -------
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x615726]

goroutine 358 [running]:
net.networkNumberAndMask(0x0, 0xc15bfea9a9fe5bc1, 0x3, 0xc00158e660, 0xc000adb270, 0x616539, 0xc00158e2ab)
	/opt/goenv/versions/1.13.0/src/net/ip.go:473 +0x26
net.(*IPNet).Contains(0x0, 0xc00158e660, 0x10, 0x10, 0x10)
	/opt/goenv/versions/1.13.0/src/net/ip.go:497 +0x2f
github.com/terraform-providers/terraform-provider-aws/aws.xmlConfigToTunnelInfo(0xc000e81000, 0xe73, 0x0, 0x0, 0xa1a56e0, 0x0, 0x0)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_vpn_connection.go:598 +0x1c3
github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsVpnConnectionRead(0xc0000fd0a0, 0x4c5fe80, 0xc0000ec000, 0x0, 0x0)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_vpn_connection.go:450 +0xe30
github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsVpnConnectionCreate(0xc0000fd0a0, 0x4c5fe80, 0xc0000ec000, 0x2, 0xa1a56e0)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_vpn_connection.go:330 +0x7fc
```

Reply:
Not sure how to elegantly catch that.. Any hints?

Reply:
Right - Got it. `tunnel1_inside_cidr` is optional. So checking for nil should do the trick..